### PR TITLE
Fix [Vehicle Spawn #115] + Fix dropPoint to Default to spawn or coords

### LINKED ISSUE #115: Vehicle Spawn
### Summary

 vehicle spawn top of another vehicle 

### Reproduction

1 try to spawn a vehicle 
2 try  to spawn another vehicle while 1st vehicle is spawned

### Expected behavior

it should do a notify    area busy  notify or spawn that vehicle side or in forward vector places notified

### Actual behavior

so what i was doing while testing the garage spawned a vehicle and tried another vehicle spawn while 1st vehicle was spawned 

### Additional context

_No response_

### Current Version

1.0.1

### Custom Resources

no

### PR DESCRIPTION
**Pull request by RoyaleWind**
## Description

Fixes #115 

## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

File: client/main.lua [hunk 1]
```diff
@@ -274,6 +274,7 @@ end
 ---@param accessPointIndex integer
 local function createZones(garageName, garage, accessPoint, accessPointIndex)
     CreateThread(function()
+        accessPoint.dropPoint = (accessPoint.dropPoint == nil) and accessPoint.spawn or accessPoint.dropPoint
```
Comment (mafewtm) [inline client/main.lua:277]:
```suggestion
        accessPoint.dropPoint = accessPoint.dropPoint or accessPoint.spawn
```

File: server/spawn-vehicle.lua [hunk 2]
```diff
@@ -22,6 +22,12 @@ lib.callback.register('qbx_garages:server:spawnVehicle', function (source, vehic
     end
     local garageType = GetGarageType(garageName)
 
+    local spawnCoords = accessPoint.spawn or accessPoint.coords
```
Comment (mafewtm) [inline server/spawn-vehicle.lua:25]: the spawnCoords variable is defined both here and line 42

File: server/spawn-vehicle.lua [hunk 3]
```diff
@@ -22,6 +22,12 @@ lib.callback.register('qbx_garages:server:spawnVehicle', function (source, vehic
...
+    if lib.getClosestVehicle(vec3(spawnCoords.x,spawnCoords.y,spawnCoords.z), 5.0, false) ~= nil then
+        exports.qbx_core:Notify(source, locale('error.no_space'), 'error', 5000)
+        return
+    end
+
```
Comment (mafewtm) [inline server/spawn-vehicle.lua:30]:
```suggestion
    local nearbyVehicle = lib.getClosestVehicle(vec3(spawnCoords.x,spawnCoords.y,spawnCoords.z), 5.0, false)

    if nearbyVehicle then
        exports.qbx_core:Notify(source, locale('error.no_space'), 'error')
        return
    end
```

File: server/spawn-vehicle.lua [hunk 4]
```diff
@@ -22,6 +22,13 @@ lib.callback.register('qbx_garages:server:spawnVehicle', function (source, vehic
     end
     local garageType = GetGarageType(garageName)
 
+    local spawnCoords = accessPoint.spawn or accessPoint.coords
+    local nearbyVehicle = lib.getClosestVehicle(vec3(spawnCoords.x, spawnCoords.y, spawnCoords.z), 5.0, false)
```
Comment (solareon) [inline server/spawn-vehicle.lua:26]: Can you pull the distance check variable out to the config. Depending on where folks place these spots 5.0 might not work for them.

**Review by solareon — CHANGES_REQUESTED** [review]

File: config/server.lua [hunk 5]
```diff
@@ -2,7 +2,7 @@ return {
     autoRespawn = false, -- True == auto respawn cars that are outside into your garage on script restart, false == does not put them into your garage and players have to go to the impound
     warpInVehicle = false, -- If false, player will no longer warp into vehicle upon taking the vehicle out.
     doorsLocked = true, -- If true, the doors will be locked upon taking the vehicle out.
-
+    distanceCheck = 5.0, -- The distance that needs to bee clear to let the vehicle spawn, this prevents staking
```
Comment (solareon) [inline config/server.lua:5]:
```suggestion
    distanceCheck = 5.0, -- The distance that needs to bee clear to let the vehicle spawn, this prevents vehicles stacking on top of each other
```

**Review by solareon — APPROVED** [review]

**Review by Manason — APPROVED** [review]

**Pull request merged**